components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandlerTest.java

```diff
@@ -928,6 +933,88 @@ public void testHandleCustomClaimsWithOAuthTokenReqMsgCtxtWithPunctuationMarkInO
 
     }
 
+    @Test
+    public void testHandleClaimsForOAuthAuthzReqMessageContextNullAccessTokenWithException() throws Exception {
+
+        JWTClaimsSet.Builder jwtClaimsSetBuilder = new JWTClaimsSet.Builder();
+
+        AuthenticatedUser authenticatedUser = getDefaultAuthenticatedLocalUser();
+        OAuth2AuthorizeReqDTO authorizeReqDTO = new OAuth2AuthorizeReqDTO();
+        authorizeReqDTO.setUser(authenticatedUser);
+        authorizeReqDTO.setTenantDomain(TENANT_DOMAIN);
+
+        OAuthAuthzReqMessageContext authzReqMessageContext = new OAuthAuthzReqMessageContext(authorizeReqDTO);
+        authzReqMessageContext.setApprovedScope(APPROVED_SCOPES);
+
+        ServiceProvider serviceProvider = new ServiceProvider();
+        serviceProvider.setApplicationName(SERVICE_PROVIDER_NAME);
+        ClaimMapping claimMap1 =
+                ClaimMapping.build("http://www.wso2.org/claims/email", "email", "sample@abc.com", true);
+        ClaimMapping claimMap2 =
+                ClaimMapping.build("http://www.wso2.org/claims/username", "username", "user123", true);
+
+        ClaimMapping[] requestedLocalClaimMap = {claimMap1, claimMap2};
+
+        ClaimConfig claimConfig = new ClaimConfig();
+        claimConfig.setClaimMappings(requestedLocalClaimMap);
+        serviceProvider.setClaimConfig(claimConfig);
+        serviceProvider.setSpProperties(new ServiceProviderProperty[]{});
+
+        OAuthServerConfiguration mockOAuthServerConfiguration = PowerMockito.mock(OAuthServerConfiguration.class);
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockOAuthServerConfiguration);
+        when(mockOAuthServerConfiguration.getOpenIDConnectSkipeUserConsentConfig()).thenReturn(true);
+
+        mockApplicationManagementService(serviceProvider);
+
+        try {
+            JWTClaimsSet jwtClaimsSet = defaultOIDCClaimsCallbackHandler.handleCustomClaims(jwtClaimsSetBuilder,
+                    authzReqMessageContext);
+            assertEquals(jwtClaimsSet.getClaims().size(), 0, "Claims are not successfully set.");
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), "Error occurred while getting claims for user " +
```

Comment:
What is the expected exceptions? If this testcase is written to handle exception, we can add ExpectedException in the testcase without try catch and assert it.